test/core.test.js

```diff
@@ -703,6 +703,36 @@ describe('core', () => {
       })
     })
 
+    describe('[Symbol.asyncIterator]', () => {
+      it('should iterate over lines from stdout', async () => {
+        const process = $`echo "Line1\nLine2\nLine3"`
+    
+        const lines = []
+        for await (const line of process) {
+          lines.push(line)
+        }
+    
+        assert.equal(lines.length, 3, 'Should have 3 lines')
+        assert.equal(lines[0], 'Line1', 'First line should be "Line1"')
+        assert.equal(lines[1], 'Line2', 'Second line should be "Line2"')
+        assert.equal(lines[2], 'Line3', 'Third line should be "Line3"')
+      });
+    
+      it('should handle partial lines correctly', async () => {
+        const process = $`node -e "process.stdout.write('PartialLine1\\nLine2\\nPartial'); setTimeout(() => process.stdout.write('Line3\\n'), 100)"`
+    
+        const lines = [];
+        for await (const line of process) {
+          lines.push(line)
+        }
+    
+        assert.equal(lines.length, 3, 'Should have 3 lines')
+        assert.equal(lines[0], 'PartialLine1', 'First line should be "PartialLine1"')
+        assert.equal(lines[1], 'Line2', 'Second line should be "Line2"')
+        assert.equal(lines[2], 'PartialLine3', 'Third line should be "PartialLine3"')
+      })
+    })
+
```

Comment:
Lets add test for empty and single line (without newline char at the end) stdouts.

Reply:
Done